datafusion/src/datasource/object_store/hdfs/os_parquet.rs

```diff
@@ -0,0 +1,102 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Parquet related things for hdfs object store
+use std::io::{Read, Seek, SeekFrom};
+use std::sync::Arc;
+
+use parquet::file::reader::{ChunkReader, Length};
+use parquet::util::io::{FileSource, TryClone};
+
+use crate::datasource::object_store::hdfs::{to_error, HadoopFileSystem};
+use crate::datasource::object_store::ObjectReader;
+use crate::datasource::parquet::dynamic_reader::{DynChunkReader, DynRead};
+
+use super::HadoopFile;
+use crate::error::Result;
+
+/// Parquet file reader for hdfs object store, by which we can get ``DynChunkReader``
+#[derive(Clone)]
+pub struct HadoopParquetFileReader {
```

Comment:
yes, our current implementation of file formats work with a generic objectstore trait object, for example: https://github.com/apache/arrow-datafusion/blob/831e07debc4f136f2e47e126b20e441f7606bd74/datafusion/src/datasource/file_format/csv.rs#L98

Reply:
Hi @alamb, I've done the refactoring. Where should I create the PR for? Directly override the PR here or create another one? And as the discussion of #907, a more preferable way for the community may be to put connectors such as S3, HDFS in their own repositories for fast development iterations. Should I create another repository for this HDFS support.

Reply:
HI @yahoNanJing  -- if you have time here is what I recommend:
1. Make a draft PR to this (arrow-datafusion) repo (so the conversation can include the code)
2. Leave a note in the PR's description that you are looking for feedback about where to put the connector (this repo or a separate one)
3. Then send a note to the dev@arrow.apache.org mailing list (or I can do this too) with a reference to the PR asking if anyone has feedback.

I have my own opinions on this matter, but I think we should get broader input before making a decision

Reply:
Hi @alamb, a new PR #1223 is created as a refactor of this PR.

Actually, in my option, it's better for DataFusion at least to support one remote object store by default in its own repository. In many companies around me, they still use HDFS. Therefore, it would be good to add HDFS support by default.

Reply:
And now the interfaces of the object store is still not stable. If one remote object store implementation is added, it will also be helpful for the interface refactoring.